File: src/main/java/com/aleosiss/diageticui/DiageticUI.java
package com.aleosiss.diageticui;

import com.aleosiss.diageticui.network.NetworkService;
import net.fabricmc.api.ModInitializer;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class DiageticUI implements ModInitializer {
    private static final Logger LOGGER = LogManager.getLogger();

    public static final String MOD_ID = "diageticui";

    @Override
    public void onInitialize() {
        LOGGER.info("main init");
        NetworkService.getInstance().initializeServer();
    }
}

File: src/main/java/com/aleosiss/diageticui/DiageticUI.kt
package com.aleosiss.diageticui

import com.aleosiss.diageticui.network.NetworkService
import net.fabricmc.api.ModInitializer
import net.minecraft.util.math.BlockPos
import org.apache.logging.log4j.LogManager

class DiageticUI : ModInitializer {
    override fun onInitialize() {
        LOGGER.info("main init")
        NetworkService.instance.initializeServer()
    }

    companion object {
        private val LOGGER = LogManager.getLogger()
        const val MOD_ID = "diageticui"
    }
}


fun Long.toBlockPos() = BlockPos.fromLong(this)
